src/main/java/com/teragrep/aer_01/DefaultOutput.java

```diff
@@ -97,17 +105,19 @@ final class DefaultOutput implements Output {
         this.resends = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "resends"));
         this.connects = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "connects"));
         this.retriedConnects = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "retriedConnects"));
-        this.sendLatency = metricRegistry.timer(name(DefaultOutput.class, "<[" + name + "]>", "sendLatency"));
-        this.connectLatency = metricRegistry.timer(name(DefaultOutput.class, "<[" + name + "]>", "connectLatency"));
+        this.sendLatency = metricRegistry.timer(name(DefaultOutput.class, "<[" + name + "]>", "sendLatency"), () -> new Timer(sendReservoir));
+        this.connectLatency = metricRegistry.timer(name(DefaultOutput.class, "<[" + name + "]>", "connectLatency"), () -> new Timer(connectReservoir));
 
         connect();
     }
 
     private void connect() {
         boolean connected = false;
         while (!connected) {
-            try (final Timer.Context context = connectLatency.time()) {
+            final Timer.Context context = connectLatency.time(); // reset the time (new context)
+            try {
                 connected = this.relpConnection.connect(relpAddress, relpPort);
+                context.close(); // manually close here, so the timer is only updated if no exceptions were thrown
```

Comment:
if line 119 connect throws, close is never called and timer is left open, will this cause a problem?

Reply:
Closing the context will always update the timed value as well. So only way not to include these cases where an exception is thrown, would be to not close it.

I looked at the source code of Timer.Context and it just calls the stop() function, which is a way of manually updating the value. It is a bit weird, stop() is documented like this: 

> Updates the timer with the difference between current and start time. Call to this method will not reset the start time. Multiple calls result in multiple updates.

And the close() is documented like this: 

> Equivalent to calling stop().

So to my knowledge it doesn't really matter if close() is called or not, the timer context is always left open.